File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/SocketFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.sjrx.intellij.plugins.systemdunitfiles.SystemdUnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.SystemdUnitFileLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class SocketFileType extends LanguageFileType {
  public static final SocketFileType INSTANCE = new SocketFileType();

  private SocketFileType() {
    super(SystemdUnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Socket unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "A unit configuration file whose name ends in \".socket\" encodes information about an IPC or network socket or a file "
           + "system FIFO controlled and supervised by systemd, for socket-based activation.";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "socket";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return SystemdUnitFileIcon.FILE;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/SocketFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class SocketFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Socket unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "socket"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.FILE
  }

  override fun getDisplayName(): @Nls String {
    return "Socket unit configuration (systemd)"
  }

  companion object {
    val INSTANCE = SocketFileType()
  }
}
